### ISSUE TITLE
RUF052: Use of dummy variables is not flagged for loop variables

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ruff#19732]
@AlexWaygood @MichaReiser Searching https://github.com/search?q=repo%3Aastral-sh%2Fruff+RUF052&type=issues does not reveal previous discussions on detecting used dummy variables inside FOR loops and list comprehensions. To avoid extra noise from rushing into opening a new issue, I would like to clarify:  Was this rejected somewhere, or was this just not discussed and not implemented yet?

Example:

```py
def no_ruff_warnings():
    my_list = [{"foo": 1}, {"foo": 2}]

    i = 0
    for _ in my_list:
        i += _["foo"]  # A dummy variable is used, but no warnings. 

    for _item in my_list:
        i += _item["foo"]  # A dummy variable is used, but no warnings. 

    [_["foo"] for _ in my_list]  # A dummy variable is used, but no warnings. 

    [_item["foo"] for _item in my_list]  # A dummy variable is used, but no warnings. 
```

_Originally posted by @alessio-locatelli in https://github.com/astral-sh/ruff/issues/14755#issuecomment-3148350171_

**Comment by MichaReiser** [comment]
I don't think this behavior is intentional. It is due to this check:

https://github.com/astral-sh/ruff/blob/331beca469edda71f214d2aa2a8b985d5641da73/crates/ruff_linter/src/rules/ruff/rules/used_dummy_variable.rs#L113-L131

The comment suggests that the intention was to only exclude parameters, but the way it is implemented excludes many other bindings too. 

We should consider making this check more fine grained (behind preview) and have a look at the ecosystem changes.

**Comment by mikeleppane** [comment]
I can take a look at this.
